src/phpDocumentor/Pipeline/Stage/Transform.php

```diff
@@ -114,35 +139,20 @@ function (PreTransformationEvent $event) : void {
         );
     }
 
-    /**
-     * @param array<int, string> $templateNames
-     */
-    private function loadTemplatesBasedOnNames(array $templateNames) : void
-    {
-        $stopWatch = new Stopwatch();
-        foreach (array_column($templateNames, 'name') as $template) {
-            $stopWatch->start('load template');
-            $this->transformer->getTemplates()->load($this->transformer, $template);
-            $stopWatch->stop('load template');
-        }
-    }
-
-    private function setTargetLocationBasedOnDsn(Dsn $dsn) : void
+    private function createFileSystem(Dsn $dsn) : FilesystemInterface
     {
         $target     = $dsn->getPath();
         $fileSystem = new Filesystem();
         if (!$fileSystem->isAbsolutePath((string) $target)) {
             $target = getcwd() . DIRECTORY_SEPARATOR . $target;
         }
 
+        $destination = $this->flySystemFactory->create(Dsn::createFromString((string) $target));
+
+        //TODO: the guides to need this, can we get rid of these lines?
```

Comment:
Perhaps if we move the creation of the flysystem to the mapping of the config to a spec; and pass that spec onto the guide renderer. But at the moment the transformer is that which binds the writers to the config